### PR TITLE
set query node thread pool size in milvus.yaml

### PR DESCRIPTION
**Pull request by cqy123456**
issue: #23120

### CODE REVIEW

File: internal/querynode/query_node.go [hunk 1]
```diff
@@ -210,8 +210,7 @@ func (node *QueryNode) InitSegcore() error {
 	C.SegcoreInit(cEasyloggingYaml)
 	C.free(unsafe.Pointer(cEasyloggingYaml))
 
-	cpuNum := runtime.GOMAXPROCS(0)
-	C.SegcoreSetThreadPoolNum(C.uint32_t(cpuNum))
+	C.SegcoreSetThreadPoolNum(Params.QueryNodeCfg.ThreadPoolSize)
```
Comment (congqixia) [inline internal/querynode/query_node.go:213]:
```suggestion
	C.SegcoreSetThreadPoolNum(C.uint32_t(Params.QueryNodeCfg.ThreadPoolSize))
```

**Comment by cqy123456** [comment]
rerun ut

File: configs/milvus.yaml [hunk 2]
```diff
@@ -200,6 +200,7 @@ queryNode:
   port: 21123
   loadMemoryUsageFactor: 3 # The multiply factor of calculating the memory usage while loading segments
   enableDisk: true # enable querynode load disk index, and search on disk index
+  ThreadPoolNumRatio: 4 # For high IOPS of disk index，thread pool size is ThreadPoolNumRatio * Number of physical cores.
```
Comment (chasingegg) [inline configs/milvus.yaml:203]: how about using lower case as other params in milvus.yaml

**Comment by cqy123456** [comment]
rerun ut

**Comment by cqy123456** [comment]
rerun ut

**Comment by cqy123456** [comment]
rerun ut

File: configs/milvus.yaml [hunk 3]
```diff
@@ -200,6 +200,7 @@ queryNode:
   port: 21123
   loadMemoryUsageFactor: 3 # The multiply factor of calculating the memory usage while loading segments
   enableDisk: true # enable querynode load disk index, and search on disk index
+  threadPoolNumRatio: 4 # For high IOPS of disk index，thread pool size is ThreadPoolNumRatio * Number of physical cores.
```
Comment (xige-16) [inline configs/milvus.yaml:203]: rename to knowherexxx would be better

**Pull request merged**